Description: flacparse: Fix resyncing after a seek
 1d913768 introduced a regression in flacparse where seeking to a previously
 unindexed position (i.e. requiring resyncing) would error out instead of
 skipping to the next start of a frame. An invalid frame would be considered as
 a valid frame of size 0 and then error out.
 .
 Backported unchanged from upstream 1.28.8 for Pi-Desktop (Ubuntu 26.04 ships
 1.28.2). Reproduced on an Orange Pi 5B with Rhythmbox's seek mode (plain
 GST_SEEK_FLAG_FLUSH): "Internal data stream error" on FLAC files without a
 SEEKTABLE.
Author: Sebastian Dröge <sebastian@centricular.com>
Origin: upstream, https://gitlab.freedesktop.org/gstreamer/gstreamer/-/commit/ef0f99387318d199d6072812c699a3fc59ac994f
Bug: https://gitlab.freedesktop.org/gstreamer/gstreamer/-/work_items/5314
Applied-Upstream: 1.28.8
Last-Update: 2026-10-08
---
diff --git a/gst/audioparsers/gstflacparse.c b/gst/audioparsers/gstflacparse.c
index 254e34cea4..801a89f015 100644
--- a/gst/audioparsers/gstflacparse.c
+++ b/gst/audioparsers/gstflacparse.c
@@ -905,7 +905,9 @@ gst_flac_parse_handle_frame (GstBaseParse * parse,
     }
 
     if (next == 0) {
-      ret = TRUE;
+      /* The frame is invalid and we have enough data to know that. Skip one
+       * byte and continue scanning for the next frame start. Don't treat it
+       * as a valid frame of size 0. */
     } else if (next > map.size) {
       GST_DEBUG_OBJECT (flacparse, "Requesting %u bytes", next);
       *skipsize = 0;
